### ISSUE TITLE
Making NPE checking as optional

### ISSUE THREAD

**Issue by GoogleCodeExporter** [randoop/randoop#20]
```
Randoop checks several generalized Java object properties and the NPE rule in 
default. However, in my experience, the NPE rule makes too many false alarms on 
both open-source projects and closed-source projects (in Google). 

While most programmers admit that NPE should be wrapped as some high-level 
exceptions, like IllegalArgumentException, they think (1) wrapping NPE as 
high-level exceptions does not make sense in most cases, particularly for 
methods that doesn't carry any state, where incorrect parameters can't "hurt" 
the next invocation, and (2) checking nullability of every reference in the 
code is not realistic for some cases. The checking code will also decrease the 
code readability. Thus, NPE is a free lunch offered by JVM to handle unexpected 
program states.

I suggest Randoop not check the NPE rule as default. Instead, we could add an 
flag to let users choose whether to check NPE or not. We could also dump all 
thrown exceptions of each method after test generation, and let users inspect 
that first. This change might make Randoop more usable.

I made the above changes in my own modified copy of Randoop (as a part of 
another tool): http://code.google.com/p/tpalus/

```

Original issue reported on code.google.com by `saizh...@google.com` on 23 Sep 2010 at 9:15

**Comment by mernst** [comment]
By "not check for NPE", I think you mean: "If the code throws a NPE, then Randoop should discard the test rather than adding it to the suite".  Is that correct?

**Comment by bjkeller** [comment]
Fixed in v2.1.0. Able to specify classification for NPE with --npe-on-null-input and --npe-on-non-null-input.
Using the value INVALID will discard any test meeting the desired condition.
